src/main/java/slimeknights/tconstruct/gadgets/item/SlimeSlingItem.java

```diff
@@ -59,22 +68,92 @@ public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity en
+          }
+
+          player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
+          SlimeBounceHandler.addBounceHandler(player);
+        }
+        break;
+      case BLUE:
+        player.addExhaustion(0.2F);
+        player.getCooldownTracker().setCooldown(stack.getItem(), 3);
+        player.setSprinting(true);
+
+        float speed = f / 3F;
+        player.addVelocity(
+          (-MathHelper.sin(player.rotationYaw / 180.0F * (float) Math.PI) * MathHelper.cos(player.rotationPitch / 180.0F * (float) Math.PI) * speed),
+          speed,
+          (MathHelper.cos(player.rotationYaw / 180.0F * (float) Math.PI) * MathHelper.cos(player.rotationPitch / 180.0F * (float) Math.PI) * speed));
+
+        if (player instanceof ServerPlayerEntity) {
+          ServerPlayerEntity playerMP = (ServerPlayerEntity) player;
+          TinkerNetwork.getInstance().sendTo(new EntityMovementChangePacket(player), playerMP);
+        }
+
+        player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
+        SlimeBounceHandler.addBounceHandler(player);
+        break;
+      case MAGMA:
+        float range = 5F;
+        Vector3d start = player.getEyePosition(1F);
+        Vector3d look = player.getLookVec();
+        Vector3d direction = start.add(look.x * range, look.y * range, look.z * range);
+        AxisAlignedBB bb = player.getBoundingBox().expand(look.x * range, look.y * range, look.z * range).expand(1, 1, 1);
+        List<Entity> entitiesInArea = player.getEntityWorld().getEntitiesWithinAABBExcludingEntity(player, bb);
+        double dist = range;
+        Entity closestEntity = null;
+        for (Entity entity : entitiesInArea) {
+          if (entity.getBoundingBox().intersects(Math.min(start.x, direction.x), Math.min(start.y, direction.y), Math.min(start.z, direction.z), Math.max(start.x, direction.x), Math.max(start.y, direction.y), Math.max(start.z, direction.z))) {
+            if (look.distanceTo(entity.getLookVec()) < dist) {
+              dist = look.distanceTo(entity.getLookVec());
+              closestEntity = entity;
+            }
+          }
+        }
+        // TODO: Ensure there isn't a block in the way
+        if (closestEntity != null) {
+          closestEntity.addVelocity(look.x * f,
+            look.y * f / 3f,
+            look.z * f);
+
+          if (closestEntity instanceof ServerPlayerEntity) {
+            ServerPlayerEntity playerMP = (ServerPlayerEntity) closestEntity;
+            TinkerNetwork.getInstance().sendTo(new EntityMovementChangePacket(closestEntity), playerMP);
+          }
+
+          player.playSound(Sounds.SLIME_SLING.getSound(), 1f, 1f);
+        }
+        break;
+      case PURPLE:
+        look = player.getLookVec();
+        double offX = look.x * f;
+        double offY = look.y * f;
+        double offZ = look.z * f;
+
+        player.setPosition(player.getPosX() + offX, player.getPosY() + offY, player.getPosZ() + offZ);
```

Comment:
Needs a teleport sound effect. Might be worth ensuring the position is valid as well (not inside a block), not sure the best way to handle that.